### DISCUSSION TITLE
How to prevent `mise use <x>` from changing global settings

### DISCUSSION THREAD

**Discussion by SamEdwardes** [jdx/mise#2303] [Q&A]
I just installed mise for the first today and so far I am loving it! Great work :) 

I have a question about how `mise use` should work.

```bash
❯ mise --version
2024.6.4 macos-arm64 (e0e4107 2024-06-15)
```

**Scenario 1 - no global config**

This works as expected. Mise use creates `.mise.toml` to save my preferences.

```bash
~/tmp/mise-test-4 
❯ rm ~/.config/mise/config.toml
rm: /Users/samedwardes/.config/mise/config.toml: No such file or directory

~/tmp/mise-test-4 
❯ ls -a
.	..

~/tmp/mise-test-4 
❯ mise use python@latest       
mise ~/tmp/mise-test-4/.mise.toml tools: python@3.12.3

~/tmp/mise-test-4 
❯ cat .mise.toml
[tools]
python = "latest"
```

**Scenario 2 - global config file**

When a global config file exists, `mise use` will override the global file, even if I don't specify `-g`. Is this the expected behavior? Is there a way I can prevent this?

```bash
~/tmp/mise-test-4 
❯ ls -a
.	..

~/tmp/mise-test-4 
❯ mise use --global python@3.12
mise ~/.config/mise/config.toml tools: python@3.12.3

# ❓ Why does this command override my global config?

~/tmp/mise-test-4 
❯ mise use python@3.11         
mise ~/.config/mise/config.toml tools: python@3.11.9

~/tmp/mise-test-4 
❯ ls -a
.	..
```

**Comment by ltsuda** [comment]
I'm having the same behaviour.

```bash
❯ mise use -g python@latest
mise ~/.config/mise/config.toml tools: python@3.12.4
...

❯ cat ~/.config/mise/config.toml
[tools]
go = "latest"
node = "lts"
python = "latest"
poetry = "latest"
usage = "latest"
...


❯ cd test

~/test
❯ ls -lah
total 8.0K
drwxr-xr-x  2 mike mike 4.0K Jul  2 23:05 .
drwxr-x--- 20 mike mike 4.0K Jul  2 23:05 ..

~/test
❯ mise use python@3.11.9
mise ~/.config/mise/config.toml tools: python@3.11.9

~/test
❯ cat ~/.config/mise/config.toml
[tools]
go = "latest"
node = "lts"
python = "3.11.9"
poetry = "latest"
usage = "latest"
```

**Reply by ltsuda** [reply]
duplicated https://github.com/jdx/mise/issues/2256

**Comment by Marukome0743** [comment]
This behavior has changed in current mise versions.

`mise use` now selects the target config using the following general order:

- `--global` updates the global config.
- `--path` explicitly selects a config file or directory.
- `--env` selects an environment-specific config.
- Otherwise, mise updates or creates a local `mise.toml`.
- When run directly in `$HOME`, it uses the global config instead of creating `$HOME/mise.toml`.

Therefore, merely having `~/.config/mise/config.toml` should no longer cause `mise use` in another project directory to overwrite the global config.

The config target resolution was centralized in https://github.com/jdx/mise/pull/6467, which was released in mise v2025.9.23.

I also verified this on the current main branch with an existing global config: running `mise use --dry-run tiny@3.0.1` from another directory selected that directory’s local `mise.toml`, not the global config.

Upgrading to a current mise release should resolve the behavior reported here.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
